text/0113-canonical-metadata.md

```diff
@@ -0,0 +1,625 @@
+                "location": { "$ref": "#/definitions/SourceLocation" },
+                "hints": { 
+                    "type": "array",
+                    "items": { "$ref": "#/definitions/DynamicImportHint" },
+                    "default": []
+                }
+            }
+        },
+        "DynamicImportHint": {
+            "type": "object",
+            "properties": {
+                "rawValue": { 
+                    "type": "string",
+                    "description": "Hint value with leading and trailing spaces of hint phrase trimmed"
+                },
+                "key": { "type": "string" },
+                "value": { "type": "string" },
+                "location": { "$ref": "#/definitions/SourceLocation" }
+            }
+        },
+        "ElementAttributeReference": {
+            "type": "object",
+            "properties": {
+                "name": { "type": "string" },
+                "value": { 
+                    "anyOf": [
+                        { "type": [ "null", "string", "boolean"] },
+                        { "$ref": "#/definitions/ComputedValue"}
+                    ]
+                }
+            }
+        },
+        "EventHandlerReference": {
+            "type": "object",
+            "properties": {
+                "name": { "type": "string" },
+                "value": { "$ref": "#/definitions/ComputedValue" }
+            }
+        },
+        "Export": {
+            "type": "object",
+            "properties": {
+                "exportsList": { 
+                    "type": "array",
+                    "items": {
+                        "type": "object",
+                        "properties": {
+                            "type": {
+                                "type": "string",
+                                "enum": ["class", "function", "expression"]
+                            }
+                        }
+                    }
+                },
+                "default": { 
+                    "type": "object",
+                    "properties": {
+                        "type": {
+                            "type": "string",
+                            "enum": ["class", "function", "expression"]
+                        }
+                    }
+                }
+            }
+        },
+        "FileType": {
+            "type": "string",
+            "enum": [ "html", "js", "css" ]
+        },
+        "Import": {
+            "type": "object",
+            "properties": {
+                "importType": { 
+                    "type": "array",
+                    "items": {
+                        "type": "string",
+                        "enum": [ "DefaultBinding", "NamedImports", "NamespacedImport"]
+                    }
+                },
+                "importsList": { "type": [ "array", "string"] },
+                "moduleSpecifier": { "$ref": "#/definitions/ModuleReference" }
```

Comment:
We should add `Location` to all objects referencing the source code of the module. This includes the `Import` type.